sql/core/src/test/resources/sql-tests/inputs/postgreSQL/groupingsets.sql

```diff
@@ -0,0 +1,563 @@
+-- select sum(c) from gstest2
+--   group by grouping sets((), grouping sets((), grouping sets(())))
+--   order by 1 desc;
+-- select sum(c) from gstest2
+--   group by grouping sets((), grouping sets((), grouping sets(((a, b)))))
+--   order by 1 desc;
+-- select sum(c) from gstest2
+--   group by grouping sets(grouping sets(rollup(c), grouping sets(cube(c))))
+--   order by 1 desc;
+-- select sum(c) from gstest2
+--   group by grouping sets(a, grouping sets(a, cube(b)))
+--   order by 1 desc;
+-- select sum(c) from gstest2
+--   group by grouping sets(grouping sets((a, (b))))
+--   order by 1 desc;
+-- select sum(c) from gstest2
+--   group by grouping sets(grouping sets((a, b)))
+--   order by 1 desc;
+-- select sum(c) from gstest2
+--   group by grouping sets(grouping sets(a, grouping sets(a), a))
+--   order by 1 desc;
+-- select sum(c) from gstest2
+--   group by grouping sets(grouping sets(a, grouping sets(a, grouping sets(a), ((a)), a, grouping sets(a), (a)), a))
+--   order by 1 desc;
+-- select sum(c) from gstest2
+--   group by grouping sets((a,(a,b)), grouping sets((a,(a,b)),a))
+--   order by 1 desc;
+
+-- empty input: first is 0 rows, second 1, third 3 etc.
+select a, b, sum(v), count(*) from gstest_empty group by grouping sets ((a,b),a);
+-- [SPARK-29701] Different answers when empty input given in GROUPING SETS
+select a, b, sum(v), count(*) from gstest_empty group by grouping sets ((a,b),());
+select a, b, sum(v), count(*) from gstest_empty group by grouping sets ((a,b),(),(),());
+select sum(v), count(*) from gstest_empty group by grouping sets ((),(),());
+
+-- empty input with joins tests some important code paths
+-- [SPARK-29701] Different answers when empty input given in GROUPING SETS
+select t1.a, t2.b, sum(t1.v), count(*) from gstest_empty t1, gstest_empty t2
+ group by grouping sets ((t1.a,t2.b),());
+
+-- simple joins, var resolution, GROUPING on join vars
+-- [SPARK-29698] Support grouping function with multiple arguments
+-- select t1.a, t2.b, grouping(t1.a, t2.b), sum(t1.v), max(t2.a)
+select t1.a, t2.b, grouping(t1.a), grouping(t2.b), sum(t1.v), max(t2.a)
+  from gstest1 t1, gstest2 t2
+ group by grouping sets ((t1.a, t2.b), ());
+
+-- [SPARK-29698] Support grouping function with multiple arguments
+-- select t1.a, t2.b, grouping(t1.a, t2.b), sum(t1.v), max(t2.a)
+select t1.a, t2.b, grouping(t1.a), grouping(t2.b), sum(t1.v), max(t2.a)
+  from gstest1 t1 join gstest2 t2 on (t1.a=t2.a)
+ group by grouping sets ((t1.a, t2.b), ());
+
+-- [SPARK-29698] Support grouping function with multiple arguments
+-- select a, b, grouping(a, b), sum(t1.v), max(t2.c)
+select a, b, grouping(a), grouping(b), sum(t1.v), max(t2.c)
+  from gstest1 t1 join gstest2 t2 using (a,b)
+ group by grouping sets ((a, b), ());
+
+-- check that functionally dependent cols are not nulled
+-- [SPARK-29698] Support grouping function with multiple arguments
+-- [SPARK-19842] Informational Referential Integrity Constraints Support in Spark
```

Comment:
Does `primary key` affect this query? Otherwise, we can remove `SPARK-19842` here.

Reply:
Yea, that might be yes (I didn't know this though..). Plz see:
```
postgres=# \d gstest3
              Table "public.gstest3"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 a      | integer |           |          | 
 b      | integer |           |          | 
 c      | integer |           |          | 
 d      | integer |           |          | 

postgres=# select a, d, grouping(a,b,c) from gstest3 group by grouping sets ((a,b), (a,c));
ERROR:  column "gstest3.d" must appear in the GROUP BY clause or be used in an aggregate function
LINE 1: select a, d, grouping(a,b,c) from gstest3 group by grouping ...
                  ^
postgres=# alter table gstest3 add primary key (a);
ALTER TABLE

postgres=# select a, d, grouping(a,b,c) from gstest3 group by grouping sets ((a,b), (a,c));
 a | d | grouping 
---+---+----------
 1 | 1 |        1
 2 | 2 |        1
 1 | 1 |        2
 2 | 2 |        2
(4 rows)
```

Reply:
`[SPARK-29702]` might include the issue in `[SPARK-19842]`, so `removing [SPARK-19842]` is ok to me.

Reply:
Got it. No problem. Let's keep it.

Reply:
ok, thanks for the quick response!